Rename keywords (#270)
MarketSquare/robotframework-tidy: docs/releasenotes/2.0.md

With following configuration:

```commandline
    robotidy --configure AlignVariablesSection:skip_types=dict
```

All variables except dictionaries will be aligned.

### `SplitTooLongLine` transformer <a name="split_too_long_line"></a>
Several updates to `SplitTooLongLine` transformer.

---

Basic behaviour changed and now every statement that's longer than
character limit will be split to multiple lines:

```
Keyword With Longer Name
...    ${arg1}
...    ${arg2}
...    ${arg3}
```

If you wish to use previous algorithm of feeding arguments into one line until limit set ``split_on_every_arg`` to `False`:

```
Keyword With Longer Name    ${arg1}  # let's asumme that ${arg} is right below character limit
...    ${arg2}    ${arg3}
```

---

Assignments are now split (if they don't fit in one line):

```
${first_assignment}
...    ${second_assignment}
...    ${third_assignment}
...    Some Lengthy Keyword So That This Line Is Too Long And Bit Over
```

### `RenameKeywords` transformer <a name="split_too_long_line"></a>

> Library names are now by default ignored by `RenameKeywords` transformer. Previously it could result in unexpected behaviour ([#269](https://github.com/MarketSquare/robotframework-tidy/issues/269)).

Reviewer: ```suggestion
Library names are now by default ignored by `RenameKeywords` transformer. Previously, it could result in an unexpected behaviour ([#269](https://github.com/MarketSquare/robotframework-tidy/issues/269)).
```

If you want to transform library names pass `ignore_library=False` to transformer.

```
    custom_library.Keyword  # custom_library will be renamed to Custom Library only if you set `ignore_library` to False
    Other Keyword
```

## Other <a name="other"></a>
### Overwrite files with `--check` flag <a name="check"></a>
It is now possible to force overwrite with ``--check`` and ``--overwrite`` flags ([#239](https://github.com/MarketSquare/robotframework-tidy/issues/239)).
Going over possible options:

Transform the files and return 0 even if there were files transformed.
```commandline
robotidy src
```

Don't transform the files and return 0 even if there would be files transformed
```commandline
robotidy --no-overwrite src
```

Don't transform the files and return 1 if there would be files transformed
```commandline
robotidy --check src
```

Transform the files and return 1 if there are files transformed
```commandline
robotidy --check --overwrite src
```

### User-friendly exceptions <a name="exceptions"></a>
Changed exceptions to be more user-friendly. Robotidy will try to recognize common mistakes.

> For unknown issues there is extra message with link to our bug board ([#250](https://github.com/MarketSquare/robotframework-tidy/issues/250))

Reviewer: ```suggestion
For unknown issues there is an extra message with link to our bug board ([#250](https://github.com/MarketSquare/robotframework-tidy/issues/250))
```

> Previously typos or common mistakes resulted in error stack being printed - often without clear information what went wrong. 

Reviewer: ```suggestion
Previously, typos or common mistakes resulted in an error stack being printed - often without clear information what went wrong. 
```

> Now `Robotidy` tries to output reason for failure:

Reviewer: ```suggestion
Now `Robotidy` tries to output the reason of a failure:
```
